vars/infra.groovy

```diff
@@ -18,3 +18,63 @@ Object withDockerCredentials(Closure body) {
         echo 'Cannot use Docker credentials outside of the trusted environment'
     }
 }
+
+void stashJenkinsWar(jenkins, stashName = "jenkinsWar") {
+    def isVersionNumber = (jenkins =~ /^\d+([.]\d+)*$/).matches()
```

Comment:
It won't work for timestamped snapshots, because they have dashes

Reply:
`isVersionNumber` applies when you have specified a released jenkins version downloadable from public maven repo (as is downloaded via maven dependency plugin) in your metadata file my understanding is that snapshots are not typically deployed there and hence if you want to use one you need a local war file or a complete URL.

@oleg-nenashev Please tell me if you foresee any problem with that approach

Reply:
Actually we deploy snapshots sometimes. But yes, it's fine to keep it as is for now. Multi-module repos have bigger problems with snapshots